pkg/util/provider/machinecontroller/machine_util.go

```diff
@@ -56,6 +60,12 @@ var (
 	emptyMap = make(map[string]string)
 )
 
+const (
+	maxReplacements    = 1
+	lockAcquireTimeout = 1 * time.Second
```

Comment:
I have tested with 100 node cluster(with around 5 machineDeployment) and no timeout log was found, so I think its sufficient. One reason is because we have lock per machineDeployment and number of worker goroutines are 50 , so there are not more than 50 goroutines waiting to acquire lock per machineDeployment in worst case.
I'll test this worst case again and will post the observation here.